### PR TITLE
Internal: Remove originalSettings from Node

### PR DESCRIPTION
**Pull request by rjernst**
This commit removes the originalSettings member from Node. It was only
needed to allows test clusters to recreate the node in certain
situations. Instead, the test cluster now keeps track of these settings.

### CODE REVIEW

**Comment by rjernst** [comment]
@elasticmachine test this please

File: test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java [hunk 1]
```diff
@@ -821,15 +822,17 @@ public synchronized void close() {
 
     private final class NodeAndClient implements Closeable {
         private MockNode node;
+        private Settings originalNodeSettings;
```
Comment (danielmitterdorfer) [inline test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java]: I think these can be declared `final`?

**Review by danielmitterdorfer — APPROVED** [review]
Looks fine overall, I left one comment. IMHO there's no further review round required so feel free to merge after addressing it.

**Comment by rjernst** [comment]
@elasticmachine test this please

**Pull request merged**